File: src/omnia/data/structure/observable/writable/WritableObservableGraph.java
package omnia.data.structure.observable.writable;

import omnia.data.contract.Writable;
import omnia.data.structure.mutable.MutableGraph;
import omnia.data.structure.observable.ObservableGraph;

public interface WritableObservableGraph<E>
    extends MutableGraph<E>, ObservableGraph<E>, Writable<ObservableGraph<E>> {}

File: src/omnia/data/structure/observable/writable/WritableObservableGraph.kt
package omnia.data.structure.observable.writable

import omnia.data.contract.Writable
import omnia.data.structure.mutable.MutableGraph
import omnia.data.structure.observable.ObservableGraph

interface WritableObservableGraph<E> : MutableGraph<E>, ObservableGraph<E>, Writable<ObservableGraph<E>>